core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -433,6 +433,6 @@ static TableMetadata fromJson(FileIO io, String metadataLocation, JsonNode node)
     return new TableMetadata(metadataLocation, formatVersion, uuid, location,
         lastSequenceNumber, lastUpdatedMillis, lastAssignedColumnId, currentSchemaId, schemas, defaultSpecId, specs,
         lastAssignedPartitionId, defaultSortOrderId, sortOrders, properties, currentVersionId,
-        snapshots, entries.build(), metadataEntries.build());
+        snapshots, entries.build(), metadataEntries.build(), ImmutableList.of() /* no changes from the file */);
```

Comment:
Changes are not written to metadata JSON files. This is how the set of changes is reset each successful commit.